File: app/src/main/java/com/futsch1/medtimer/statistics/ActiveStatisticsFragment.java
package com.futsch1.medtimer.statistics;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.preference.PreferenceManager;

public class ActiveStatisticsFragment {
    private static final int DEFAULT_FRAGMENT = StatisticFragmentType.CHARTS.ordinal();
    private final SharedPreferences sharedPref;

    public ActiveStatisticsFragment(Context context) {
        sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
    }

    public StatisticFragmentType getActiveFragment() {
        return StatisticFragmentType.values()[sharedPref.getInt("active_statistics_fragment", DEFAULT_FRAGMENT)];
    }

    public void setActiveFragment(StatisticFragmentType fragment) {
        sharedPref.edit().putInt("active_statistics_fragment", fragment.ordinal()).apply();
    }

    public enum StatisticFragmentType {
        CHARTS,
        TABLE,
        CALENDAR
    }
}

File: app/src/main/java/com/futsch1/medtimer/statistics/ActiveStatisticsFragment.kt
package com.futsch1.medtimer.statistics

import android.content.Context
import android.content.SharedPreferences
import androidx.core.content.edit
import androidx.preference.PreferenceManager
import com.futsch1.medtimer.preferences.PreferencesNames.ACTIVE_STATISTICS_FRAGMENT

class ActiveStatisticsFragment(context: Context) {

    companion object {
        private val DEFAULT_FRAGMENT = StatisticFragmentType.CHARTS.ordinal
    }

    private val sharedPref: SharedPreferences =
        PreferenceManager.getDefaultSharedPreferences(context)

    var activeFragment: StatisticFragmentType
        get() = StatisticFragmentType.entries[sharedPref.getInt(
            ACTIVE_STATISTICS_FRAGMENT,
            DEFAULT_FRAGMENT
        )]
        set(fragment) {
            sharedPref.edit { putInt(ACTIVE_STATISTICS_FRAGMENT, fragment.ordinal) }
        }
}
